tutorials/appengine-serve-machine-learning-model/index.md

```diff
@@ -0,0 +1,176 @@
+
+- The model needs to process requests in real time when the users are interacting with your web application while at the same time batch process interaction logs you have stored previously in a database.
+
+- The model's output is used by multiple other machine learning models in your application.
+
+[App Engine][appengine] offers rolling update, networking, and auto scaling.
+
+[Cloud Endpoints][endpoints] helps you monitor and manage permission and quota of the service's consumers.
+
+You can follow the [steps of the sample app][steps] to deploy a service.  Below we will look at some key pieces of the code to understand how it works.
+
+## A closer look
+
+### [`main.py`][main.py]
+
+Our app expects `POST` requests to the path `/predict`.  It will look for the value of `'X'` in the JSON data, send it to the trained model, and return the result as the value of `'y'`:
+
+```python
+@app.route('/predict', methods=['POST'])
+def predict():
+    X = request.get_json()['X']
+    y = MODEL.predict(X).tolist()
+    return json.dumps({'y': y}), 200
+```
+
+Here `MODEL` is a global variable for the trained machine learning model we are serving.  To make sure the model is loaded, we use the `before_first_request` decorator, which will be triggered by App Engine's health check requests:
+
+```python
+MODEL_BUCKET = os.environ['MODEL_BUCKET']
+MODEL_FILENAME = os.environ['MODEL_FILENAME']
+MODEL = None
+
+@app.before_first_request
+def _load_model():
+    global MODEL
+    client = storage.Client()
+    bucket = client.get_bucket(MODEL_BUCKET)
+    blob = bucket.get_blob(MODEL_FILENAME)
+    s = blob.download_as_string()
+
+    MODEL = pickle.loads(s)
+```
+
+We store the model as a pickled file on [Cloud Storage][storage].  To make sure the App Engine service knows where to find the model, we pass in the model's bucket and file name as environment variables.  This is done in the `app.yaml` configuration file.
+
+
+### [`app.yaml`][app.yaml]
+
+The `app.yaml` configuration file defines the App Engine service.  We define environment variables pointing to the trained model:
+
+```yaml
+env_variables:
+    # The app will look for the model file at: gs://MODEL_BUCKET/MODEL_FILENAME
+    MODEL_BUCKET: BUCKET_NAME
+    MODEL_FILENAME: lr.pkl
+```
+
+You should replace `BUCKET_NAME` with a bucket owned by your project.  `lr.pkl` is a simple linear regression model included in the sample app.
+
+To use Cloud Endpoints to manage the service, we need to specify a `config_id` under the `endpoints_api_service` field:
+
+```yaml
+endpoints_api_service:
+  name: modelserve-dot-PROJECT_ID.appspot.com
+  config_id: CONFIG_ID
+```
+
+The `CONFIG_ID` points to a Cloud Endpoints service deployment.  You can find all the deployments on the [Cloud Endpoints console][endpoints] under each service page's Deployment history tab.
```

Comment:
Can you be more specific about which field in the Cloud Endpoints console contains the value you want entered for CONFIG_ID?

Reply:
Added more info here, please take a look.